tests/basic/storage_pool_test.go

```diff
@@ -11023,3 +11023,133 @@ var _ = Describe("{CheckPoolOffline}", func() {
+		    4. The request must be rejected and the journal device should not be added in lsblk
+		    5. Now put pool in maintenance mode and add a cloud drive as a journal device this should go through successfully
+	*/
+
+	JustBeforeEach(func() {
+		StartTorpedoTest("OnlineJournalAddCheck", "Online journal add check", nil, 0)
+	})
+
+	var contexts []*scheduler.Context
+	var selectedNode = node.Node{}
+	var lsblkOutput string
+
+	itLog := "OnlineJournalAddCheck"
+	It(itLog, func() {
+		log.InfoD(itLog)
+		stepLog := "Schedule application"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			for i := 0; i < Inst().GlobalScaleFactor; i++ {
+				contexts = append(contexts, ScheduleApplications(fmt.Sprintf("online-journal-add-check-%d", i))...)
+			}
+		})
+		ValidateApplications(contexts)
+		defer DestroyApps(contexts, nil)
+
+		stepLog = "select a node where there is no journal device"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			storageNodes := node.GetStorageDriverNodes()
+			for _, node := range storageNodes {
+				output, err := runCmd("pxctl status", node)
+				log.FailOnError(err, "Failed to run pxctl status on node: %v", node.Name)
+				log.Infof("pxctl status output: %v", output)
+
+				if !strings.Contains(output, "Journal Device") {
+					selectedNode = node
+					break
+				}
+			}
+			if selectedNode.Name == "" {
+				log.FailOnError(fmt.Errorf("No node found without journal device"), "No node found without journal device")
+			}
+			log.Infof("Selected Node for add journal device: %v", selectedNode.Name)
+		})
+		stepLog = "Check lsblk state before addition of drive add of journal device"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			//run lsblk command to check the state of the journal device
+			cmd := "lsblk"
+			lsblkOutput, err = runCmd(cmd, selectedNode)
+			log.FailOnError(err, "Failed to run command: %v", cmd)
+			log.InfoD("lsblk output: %v", lsblkOutput)
+
+		})
+
+		stepLog = "Add a cloud drive as a journal device without pool in maintenance mode"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			//Add a cloud drive as a journal device without pool in maintenance mode
+			driveSpecs, err := GetCloudDriveDeviceSpecs()
+			log.FailOnError(err, "Error getting cloud drive specs")
+
+			deviceSpec := driveSpecs[0]
+			devicespecjournal := deviceSpec + " --journal"
+
+			err = Inst().V.AddCloudDrive(&selectedNode, devicespecjournal, -1)
+			if err == nil {
+				log.FailOnError(fmt.Errorf("adding cloud drive with journal expected ? Error: [%v]", err),
+					"adding cloud drive with journal failed ?")
+			}
+			log.InfoD("adding journal failed as expected. verifying the error: %v", err.Error())
+
+			re := regexp.MustCompile(".*Requires pool maintenance mode*")
+			if !re.MatchString(err.Error()) {
+				log.FailOnError(err, "Requires pool maintenance mode alert is not received")
+			}
+			//Check lsblk output this should be same as the old output
+			time.Sleep(30 * time.Second)
```

Comment:
Yeah this is required because I have raised a bug PWX-36300 what happens here is first it is adding drives and then checking if pool is in maintenance mode and for a brief while there are new entries in lsblk, hence for it catch up I am giving it 30 seconds as of now. Maybe we can remove this once the bug is fixed.